### PR TITLE
update reaper, scaler, describer for GroupKind

### PR DESCRIPTION
**Pull request by deads2k**
Updating reaper, scaler, and describer lookups to use `GroupKinds`.

@liggitt @caesarxuchao

### CODE REVIEW

File: pkg/api/register.go [hunk 1]
```diff
@@ -27,6 +27,10 @@ var Scheme = runtime.NewScheme()
 // SchemeGroupVersion is group version used to register these objects
 var SchemeGroupVersion = unversioned.GroupVersion{Group: "", Version: ""}
 
+func Kind(kind string) unversioned.GroupKind {
```
Comment (derekwaynecarr) [inline pkg/api/register.go]: Godoc please

File: pkg/apis/componentconfig/register.go [hunk 2]
```diff
@@ -28,6 +28,10 @@ func init() {
 // SchemeGroupVersion is group version used to register these objects
 var SchemeGroupVersion = unversioned.GroupVersion{Group: "componentconfig", Version: ""}
 
+func Kind(kind string) unversioned.GroupKind {
```
Comment (derekwaynecarr) [inline pkg/apis/componentconfig/register.go]: Godoc please

Comment (derekwaynecarr) [inline pkg/apis/componentconfig/register.go]: abstain from commenting on same throughout

**Comment by derekwaynecarr** [comment]
nits, lgtm

File: pkg/kubectl/describe.go [hunk 3]
```diff
@@ -107,17 +105,8 @@ func DescribableResources() []string {
...
-
-	switch group {
-	case "":
-		f, ok = describerMap(c)[kind]
-	case "extensions":
-		f, ok = expDescriberMap(c)[kind]
-	}
-
+func DescriberFor(kind unversioned.GroupKind, c *client.Client) (Describer, bool) {
+	f, ok := describerMap(c)[kind]
```
Comment (soltysh) [inline pkg/kubectl/describe.go]: `return describerMap(c)[kind]`

Comment (deads2k) [inline pkg/kubectl/describe.go]: > return describerMap(c)[kind]

That won't compile.  the `value, ok` return is some sort of special cased thing.  When done with a return, its just the `value`

**Comment by deads2k** [comment]
godoc added

File: pkg/kubectl/describe.go (hunk 3 above)
Comment (soltysh) [inline pkg/kubectl/describe.go]: :disappointed_relieved:

**Comment by soltysh** [comment]
LGTM

**Comment by deads2k** [comment]
@k8s-bot unit test this please

**Comment by deads2k** [comment]
Two lgtm's, straight-forward change.  Tagging.

**Comment by wojtek-t** [comment]
@deads2k 

hack/verify-generated-docs.sh:28 is failing for this PR
Can you please fix it?

**Comment by wojtek-t** [comment]
@deads2k - one more issue I think:

06:16:44 /go/src/k8s.io/kubernetes/test/integration/framework/master_utils.go:192: cannot use "ReplicationController" (type string) as type "k8s.io/kubernetes/pkg/api/unversioned".GroupKind in argument to kubectl.ReaperFor

**Comment by deads2k** [comment]
> @deads2k - one more issue I think:

Busy busy beavers over thanksgiving :)

**Comment by wojtek-t** [comment]
I also reviewed it - thanks!

LGTM

**Pull request merged**